### PR TITLE
Clean up testrunner parameters

### PR DESCRIPTION
**Pull request by cmurphy**
The commits in this PR address technical debt in the testrunner by reorganizing parameters to be more logically grouped, as well as by removing unneeded cruft.

Fixes https://github.com/SUSE/avant-garde/issues/710

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/infra/testrunner/README.md [hunk 1]
```diff
@@ -110,9 +111,6 @@ username: "username"
 
 There are some arguments that are currently at the top level of the configuration, but are actually related to the platform:
```
Comment (tdaines42) [inline ci/infra/testrunner/README.md]: I think maybe this can be removed now

Comment (cmurphy) [reply ci/infra/testrunner/README.md]: done

File: ci/infra/testrunner/platforms/platform.py [hunk 2]
```diff
@@ -103,11 +103,11 @@ def provision(self, num_master=-1, num_worker=-1):
         if num_master > -1 or num_worker > -1:
             logger.warning("Overriding number of nodes")
             if num_master > -1:
-                self.conf.master.count = num_master
+                self.conf.terraform.master.count = num_master
```
Comment (Itxaka) [inline ci/infra/testrunner/platforms/platform.py]: oh, so this is why this wasnt working before! Nice catch!

**Comment by jodavis-suse** [comment]
LGTM, though I'm not an expert. :)

**Pull request merged**